velox/functions/sparksql/aggregates/tests/AverageAggregationTest.cpp

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/exec/tests/utils/AssertQueryBuilder.h"
+#include "velox/exec/tests/utils/PlanBuilder.h"
+#include "velox/functions/lib/aggregates/tests/AggregationTestBase.h"
+#include "velox/parse/TypeResolver.h"
+#include "velox/vector/tests/utils/VectorTestBase.h"
+
+#include "velox/functions/sparksql/aggregates/Register.h"
+
+using namespace facebook::velox::exec::test;
+using namespace facebook::velox::functions::aggregate::test;
+
+namespace facebook::velox::functions::aggregate::sparksql::test {
+
+namespace {
+
+class AverageAggregationTest : public AggregationTestBase {
+ protected:
+  void SetUp() override {
+    AggregationTestBase::SetUp();
+    allowInputShuffle();
+    registerAggregateFunctions("spark_");
+  }
+};
+
+TEST_F(AverageAggregationTest, avgAllNulls) {
+  vector_size_t size = 1'000;
+  // Have two row vectors a least as it triggers different code paths.
+  auto vectors = {
+      makeRowVector({
+          makeAllNullFlatVector<int64_t>(size),
+      }),
+      makeRowVector({
+          makeAllNullFlatVector<int64_t>(size),
+      }),
+  };
+
+  createDuckDbTable(vectors);
+
+  testAggregations(
+      vectors,
+      {},
```

Comment:
Would it be possible to add a group-by case for completeness, where some groups are all-null and some groups are not.